source/cli/ArgumentParser.ooc

```diff
@@ -48,68 +49,68 @@ ArgumentParser: class {
-		this _arguments add(_Argument new(longIdentifier claim(), shortIdentifier, 0, action))
+	add: func (longIdentifier: String, shortIdentifier: Char, action: Event) {
+		this _arguments add(_Argument new(longIdentifier, shortIdentifier, 0, action))
 	}
-	add: func ~noShort (longIdentifier: Text, action: Event) {
-		this _arguments add(_Argument new(longIdentifier claim(), 0, action))
+	add: func ~noShort (longIdentifier: String, action: Event) {
+		this _arguments add(_Argument new(longIdentifier, 0, action))
 	}
-	add: func ~parameter (longIdentifier: Text, shortIdentifier: Char, action: Event1<Text>) {
-		this _arguments add(_Argument new(longIdentifier claim(), shortIdentifier, 1, action))
+	add: func ~parameter (longIdentifier: String, shortIdentifier: Char, action: Event1<String>) {
+		this _arguments add(_Argument new(longIdentifier, shortIdentifier, 1, action))
 	}
-	add: func ~parameterNoShort (longIdentifier: Text, action: Event1<Text>) {
-		this _arguments add(_Argument new(longIdentifier claim(), 1, action))
+	add: func ~parameterNoShort (longIdentifier: String, action: Event1<String>) {
+		this _arguments add(_Argument new(longIdentifier, 1, action))
 	}
-	add: func ~parameters (longIdentifier: Text, shortIdentifier: Char, parameters: Int, action: Event1<VectorList<Text>>) {
-		this _arguments add(_Argument new(longIdentifier claim(), shortIdentifier, parameters, action))
+	add: func ~parameters (longIdentifier: String, shortIdentifier: Char, parameters: Int, action: Event1<VectorList<String>>) {
+		this _arguments add(_Argument new(longIdentifier, shortIdentifier, parameters, action))
 	}
-	add: func ~parametersNoShort (longIdentifier: Text, parameters: Int, action: Event1<VectorList<Text>>) {
-		this _arguments add(_Argument new(longIdentifier claim(), parameters, action))
+	add: func ~parametersNoShort (longIdentifier: String, parameters: Int, action: Event1<VectorList<String>>) {
+		this _arguments add(_Argument new(longIdentifier, parameters, action))
 	}
-	addDefault: func (action: Event1<Text>) {
+	addDefault: func (action: Event1<String>) {
 		this _defaultArgument = _Argument new(action)
 	}
-	parse: func (input: VectorList<Text>) {
-		parameters := VectorList<Text> new()
-		arguments := VectorList<Text> new()
+	parse: func (input: VectorList<String>) {
+		parameters := VectorList<String> new()
+		arguments := VectorList<String> new()
 		for (i in 0 .. input count) {
 			(parameters, arguments) clear()
-			current := input[i] take()
-			if (current beginsWith(t"--"))
-				arguments add(current slice(2))
-			else if (current beginsWith(t"-")) {
-				current = current slice(1)
-				for (j in 0 .. current count)
-					arguments add(current slice(j, 1))
-			}
+			current := input[i]
+			if (current startsWith("--"))
+				arguments add(current substring(2))
+			else if (current startsWith("-"))
+				for (j in 1 .. current size)
+					arguments add(current[j] as String)
```

Comment:
Because you can overload the `as` operator. So `current[j] as String` actually creates a new `String` of length 1 with `current[j]` as the only character.